kernel/kernel-impl/src/main/java/org/sakaiproject/db/impl/BasicSqlService.java

```diff
@@ -979,6 +979,13 @@ public boolean dbWriteBinary(String sql, Object[] fields, byte[] var, int offset
 		}
 		catch (SQLException e)
 		{
+			// On mysql unless you allow serverside prepared statements then the maximum size possible is configured
+			// by max_allowed_packet. The error codes below are:
+			// 1105 max_allowed_packet too small
+			// 1118 redo log size not at least 10 times max_allowed_packet
+			if ( "mysql".equals(m_vendor) && (e.getErrorCode() == 1105 || e.getErrorCode() == 1118) ) {
+				LOG.warn(String.format("SQL '%s' failed, consider useServerPrepStmts=true on JDBC connection.", sql), e);
```

Comment:
I don't think the SLF4J one works because all the arguments are format subsitutions:

    LOG.warn("SQL '{}' failed, consider useServerPrepStmts=true on JDBC connection.", sql, e);

will use:

     public void warn(String format, Object arg1, Object arg2);

and ignore the stack trace. But I'm wrong, `org.slf4j.helpers.MessageFormatter#getThrowableCandidate` gets called which extracts the throwable out of the argument array. 

This is in the FAQ, but I didn't see that when looking at the JavaDoc: http://slf4j.org/faq.html#paramException